onnxruntime/core/session/inference_session.cc

```diff
@@ -188,11 +192,23 @@ class InferenceSession::Impl {
     return Status::OK();
   }
 
-  common::Status RegisterGraphTransformer(std::unique_ptr<onnxruntime::GraphTransformer> p_graph_transformer) {
+  common::Status RegisterGraphTransformer(std::unique_ptr<onnxruntime::GraphTransformer> p_graph_transformer,
+                                          std::vector<std::string>&& providers,
+                                          TransformerLevel level = TransformerLevel::Optional_L2) {
     if (p_graph_transformer == nullptr) {
       return Status(common::ONNXRUNTIME, common::FAIL, "Received nullptr for graph transformer");
     }
-    return graph_transformation_mgr_.Register(std::move(p_graph_transformer));
+    return graph_transformation_mgr_.Register(std::move(p_graph_transformer), level, std::move(providers));
+  }
+
+  common::Status AddCustomTransformerList(std::vector<std::string>&& transformers_to_enable) {
+    if (transformers_to_enable_.empty()) {    
+      transformers_to_enable_ = std::move(transformers_to_enable);
+    } else {
+      std::move(transformers_to_enable.begin(), transformers_to_enable.end(), std::back_inserter(transformers_to_enable_));
+      transformers_to_enable.clear();
+    }
+    return Status::OK();
```

Comment:
While this is technically more efficient to run, if you just took a reference for transformers_to_enable this is a single line of code (std::copy with back_inserter). In this case simple may have been a better balance (less time to write, easier to maintain).